Cython/Compiler/PyrexTypes.py

```diff
@@ -4782,6 +4792,11 @@ def specialize_here(self, pos, env, template_values=None):
         # Replace this type with the actual 'template' argument.
         return template_values[0].resolve()
 
+class PyNoneType(BuiltinObjectType):
+
+    def __init__(self):
+        super().__init__('None', None, [])
```

Comment:
Shouldn't the name of the type be `NoneType`?
Is it ok that we don't define a `cname` here? (CPython uses the private name `_PyNone_Type`, which we certainly don't want to use.)
And why do you pass a list for the `objstruct_cname`?
```suggestion
    def __init__(self):
        super().__init__('NoneType', cname=None)
```